app/javascript/packages/decorators/once.ts

```diff
@@ -0,0 +1,12 @@
+export const once = (): MethodDecorator => <T>(_target, _propertyKey, descriptor) => {
+  const get = descriptor.get!;
+  descriptor.get = function () {
+    if (!descriptor.value) {
```

Comment:
is descriptor per-instance of this function, or is it a singular static descriptor?

Reply:
That's a good question! Without testing, I'm actually not sure. But I'll make sure to have a spec for it working the way we'd expect it to.

Reply:
ok great, this matches my understanding of these more now, glad we checked